src/components/Garden/ProposalDetail/ProposalDetail.js

```diff
@@ -388,6 +416,148 @@ function ProposalDetail({
   )
 }
 
+function ArgumentBox({ proposal }) {
+  const theme = useTheme()
+
+  const [challengeContext, setChallengeContext] = useState()
+  const [challengerAddress, setChallengerAddress] = useState()
+  const [challengerData, setChallengerData] = useState()
+  const [showArgument, setShowArgument] = useState(false)
+
+  const { agreementActions } = useActions()
+
+  useEffect(() => {
+    async function getChallengeData() {
+      const challengeData = await agreementActions.getChallenge(
+        proposal.challengeId
+      )
+      setChallengeContext(hexToUtf8(challengeData.context))
+      setChallengerAddress(challengeData.challenger)
+    }
+    getChallengeData()
+
+    async function fetchChallengerProfile() {
+      const challengerProfile = await getProfileForAccount(challengerAddress)
+      setChallengerData(challengerProfile)
+    }
+    fetchChallengerProfile()
+  }, [challengerAddress])
```

Comment:
What do you think of creating a hook `useChallenge` or similar which encapsulates this logic.

Also we could have one state variable object and set all values in one go.
```suggestion
  const [challenge, setChallenge] = useState()
  const [showArgument, setShowArgument] = useState(false)

  const { agreementActions } = useActions()

  useEffect(() => {
    async function getChallengeData() {
      const challengeData = await agreementActions.getChallenge(
        proposal.challengeId
      )
      const challengerProfile = await getProfileForAccount(
        challengeData.challenger
      )

      setChallenge({
        context: hexToUtf8(challengeData.context),
        challenger: { address: challengeData.challenger, ...challengerProfile },
      })
    }
    getChallengeData()
  }, [agreementActions, proposal.challengeId])
```